common.mk

```diff
@@ -25,8 +25,23 @@ ifeq ($(name),)
   $(error name ENV is not set)
 endif
 
+PYTHON = $(shell which python3)
+ifeq ($(PYTHON),)
+  PYTHON = $(shell which python)
+  ifeq ($(PYTHON),)
+  	$(error "PYTHON=$(PYTHON) not found in $(PATH)")
+  endif
+endif
+PYTHON_VERSION_MIN=3.0
+PYTHON_VERSION=$(shell $(PYTHON) -c \
+'import sys; print(float(str(sys.version_info[0]) + "." + str(sys.version_info[1])))')
+PYTHON_VERSION_OK=$(shell $(PYTHON) -c 'print(int($(PYTHON_VERSION) >= $(PYTHON_VERSION_MIN)))' )
+ifeq ($(PYTHON_VERSION_OK), 0) # True == 1
+   $(error "Need python version >= $(PYTHON_VERSION_MIN) (current: $(PYTHON_VERSION))")
+endif
+
 # Thx to https://stackoverflow.com/questions/5618615/check-if-a-program-exists-from-a-makefile
-EXECUTABLES = make docker kind git node npm npx kubectl helm yq java python
+EXECUTABLES = make docker kind git node npm npx kubectl helm yq java $(PYTHON)
 K := $(foreach exec,$(EXECUTABLES),\
```

Comment:
What is `K`?

Reply:
`$(info $$K is [${K}])` gives `$K is [ some string  some string  some string  some string  some string  some string  some string  some string  some string  some string  some string  some string]`

It is not a change of the PR, but the [link](https://stackoverflow.com/questions/5618615/check-if-a-program-exists-from-a-makefile) mentioned in the comment above states that "K" is necessary as an arbitrary variable name in order to have the right hand side executed immediately.

Reply:
Maybe you should choose a better name for this variable.

Reply:
It is not part of the PR, should I change it anyway?

Reply:
Yes, that's the pathfinder rule of clean code: leave the place cleaner than you found it. This rule helps to improve the whole code base step by step over time.

Reply:
@Weltraumschaf Alright I refactored the name :) If possible, could you check if it is running with PyEnv? I think you are using it..

Reply:
works:

```
» pyenv version
3.9.10 (set by /Users/sst/src/work/scb/secureCodeBox/.python-version)
...
tests/git_repo_scanner_test.py ........                                                                                                                                                                                              [100%]

============================================================================================================ 8 passed in 0.62s =============================================================================================================
```